src/main/java/org/skriptlang/skript/bukkit/whitelist/elements/CondWillBeWhitelisted.java

```diff
@@ -0,0 +1,74 @@
+package org.skriptlang.skript.bukkit.whitelist.elements;
+
+import com.destroystokyo.paper.event.server.WhitelistToggleEvent;
+import io.papermc.paper.event.server.WhitelistStateUpdateEvent;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.util.Kleenean;
+
+@Name("Will Be Whitelisted")
+@Description("Checks whether the server or a player will be whitelisted in a <a href='events.html#whitelist'>whitelist</a> event.")
+@Keywords("server, player")
+@Examples({
+	"on server whitelist:",
+		"\tsend \"Server whitelist has been set to % whether server will be whitelisted%\" to all ops",
+	"",
+	"on player whitelist:",
+		"\tsend \"Whitelist of player % event - player % has been set to % whether server will be whitelisted%\" to all ops"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper")
+public class CondWillBeWhitelisted extends Condition {
+
+	static {
+		Skript.registerCondition(CondWillBeWhitelisted.class,
+			"[the] (:player|server) will be whitelisted",
+			"[the] (:player|server) (will not|won't) be whitelisted"
+		);
```

Comment:
This should probably just be a time state of CondIsWhitelisted, but due to the added complexity I think it's fine being a separate condition.